### ISSUE TITLE
Duplicated projects should have unique names

### ISSUE THREAD

**Issue by tatiang** [thunkable/thunkable-issues#1066]
### Steps to Reproduce
1. Duplicate a project from the [Projects page](https://x.thunkable.com/projects). 
2. Duplicate the same project again.*

*I think there may have been an intermediate step between 1 and 2 that I can't recall.

### Expected Behavior
A project named "Lunch Plan v2" should have a duplicated project named "Lunch Plan v2 copy". And if I duplicate the first project again, the new duplicate should be named "Lunch Plan v2 copy 2" instead of "Lunch Plan v2 copy" which conflicts with the first duplicate.

### Actual Behavior
See above.

### Project Link

###Community Link
If there's an ongoing discussion about this behavior in the Community, please link to it here

```
  STACK TRACE AND/OR SCREENSHOTS
```
![image](https://user-images.githubusercontent.com/10355498/142820506-a7af2402-8fe7-4162-b185-075d6167afd9.png)

**Comment by tatiang** [comment]
I think it is creating unique names by appending numbers. So that's good. But it's still showing incorrect date modified values (https://github.com/thunkable/thunkable-issues/issues/1014).

**Comment by tatiang** [comment]
This issue never really went away. It works less often than it fails.

I just made a copy of the "Ant Game Helper" project. I have no idea which project is the newest copy. The last four items in the list have the same name and the same modified date & time (the original date & time). I don't even know what to do with this! Do I have to open every project and try to determine which one is the actual newest one? It makes using Thunkable really difficult.

![image](https://user-images.githubusercontent.com/10355498/147868453-cd73018f-1927-49f0-a296-0474bb2191ee.png)

**Comment by tatiang** [comment]
This continues to be a problem.

See also https://community.thunkable.com/t/bugs-copying-duplicating-projects-fails/1744947

**Comment by josmas** [comment]
Hey @tatiang, thanks for the report. The pattern I am seeing is that if I make a copy it appends the word `copy` at the end. Another copy of the same project will append a number at the end. But it does not seem to be consistent. I'll queue it in for triaging.

**Comment by tatiang** [comment]
> Hey @tatiang, thanks for the report. The pattern I am seeing is that if I make a copy it appends the word `copy` at the end. Another copy of the same project will append a number at the end. But it does not seem to be consistent. I'll queue it in for triaging.

I know that's the intended behavior but at least for me, it's wildly inconsistent. Sometimes, no copy is made. That's rare. Sometimes, a copy is made with the exact same name as an existing copy (e.g. "Project copy 2" and "Project copy 2"). And sometimes, a copy is made with the correct name (e.g. "Project copy 2" --> "Project copy 3") but with the modified date being the original creation date. That's what happens most often for me.

**Comment by josmas** [comment]
yeah, it seems to be a bit inconsistent. The modified date is a related but different issue. It is a tricky one, cause we are following the same practice as when you make a copy on mac Finder (uses the same created date as original) but in our case I am not sure it works. In any case, let's leave that discussion for the other thread. We'll treat them as two separate issues (even though they are related). Thanks!

**Comment by tatiang** [comment]
I understand, thank you.
